string filename
bool wipe
---
string error
